src/Elasticsearch.Net/Connection/HttpConnection.cs

```diff
@@ -49,21 +52,28 @@ public virtual TResponse Request<TResponse>(RequestData requestData)
 			Stream responseStream = null;
 			Exception ex = null;
 			string mimeType = null;
+			IDisposable receive = DiagnosticSources.SingletonDisposable;
 			try
 			{
 				var requestMessage = CreateHttpRequestMessage(requestData);
-				responseMessage = client.SendAsync(requestMessage).GetAwaiter().GetResult();
+				SetContent(requestMessage, requestData);
+				using(requestMessage?.Content ?? (IDisposable)Stream.Null)
+				using (var d = DiagnosticSource.Diagnose<RequestData, int?>(DiagnosticSources.HttpConnection.SendAndReceiveHeaders, requestData))
+				{
+					responseMessage = client.SendAsync(requestMessage, HttpCompletionOption.ResponseHeadersRead).GetAwaiter().GetResult();
```

Comment:
Deadlock risk, I guess there is no way to make this method async?

Reply:
:+1:No, however `HttpConnection` backed by this will only be referenced in `netstandard`. On full framework `net461` target will opt in to `HttpWebRequest` backed `HttpConnection` again when #3831 lands so in scenarios where there is a synchronization context `IConnection.Request()` will be fully synchronous.